tests/cases/conformance/es2022/es2022SharedMemory.ts

```diff
@@ -0,0 +1,9 @@
+// @target: esnext
+// @lib: es2022.sharedmemory
+// @noemit: true
+// @strict: true
+
+const sab = new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT * 1024);
+const int32 = new Int32Array(sab);
+Atomics.wait(int32, 0, 0);
+await Atomics.waitAsync(int32, 0, 0);
```

Comment:
The return value of `Atomics.waitAsync` is a `{async: boolean, value: ("ok" | "not-equal" | "timeout" | Promise<"ok" | "not-equal" | "timeout">)` so you need to wait on `value`.
```suggestion
const {async, value} = Atomics.waitAsync(int32, 0, 0);
await value;
```